clang/lib/CodeGen/BuiltinTargets/AArch64.cpp

```diff
@@ -0,0 +1,8530 @@
+//===------ AArch64.cpp - Emit LLVM Code for builtins ---------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This contains code to emit Builtin calls as LLVM code.
+//
+//===----------------------------------------------------------------------===//
+
```

Comment:
Not a strong opinion, but this appear to handle both ARM and AArch64, which is similar to `SemaARM`, so maybe this would be more consistent if named just `ARM`?

Reply:
Thanks @MacDue. It's about 90% `AArch64` code, and I did briefly look at splitting it, but it's tightly coupled. If it's okay, I'll leave it as-is since I think `AArch64.cpp` better explains the contents. I might try to split it in future, but I've got to merge #132060 before I can land this PR, and I'd like to get this merged soon since it would reduce my compile times.

Reply:
Sure, no worries :+1:  (just an observation -- not blocking this PR).  It may be worth noting that this file contains some ARM builtins too in the comment at the top, just to make it clear they're slightly intertwined. 

Eg. at the:
> This contains code to emit Builtin calls as LLVM code.

Reply:
Now done.